crates/sui-move-build/src/lib.rs

```diff
@@ -129,52 +119,28 @@ impl BuildConfig {
         let mut fn_info = None;
         let compiled_pkg = build_plan.compile_with_driver(writer, |compiler| {
             let (files, units_res) = if lint {
-                let lint_visitors = vec![
-                    ShareOwnedVerifier.visitor(),
-                    SelfTransferVerifier.visitor(),
-                    CustomStateChangeVerifier.visitor(),
-                    CoinFieldVisitor.visitor(),
-                    FreezeWrappedVisitor.visitor(),
-                    CollectionEqualityVisitor.visitor(),
-                ];
                 let (filter_attr_name, filters) = known_filters();
                 compiler
-                    .add_visitors(lint_visitors)
+                    .add_visitors(linter_visitors())
                     .add_custom_known_filters(filters, filter_attr_name)
                     .build()?
             } else {
                 compiler.build()?
             };
             match units_res {
                 Ok((units, warning_diags)) => {
-                    let any_linter_warnings = warning_diags.any_with_prefix(LINT_WARNING_PREFIX);
-                    let (filtered_diags_num, filtered_categories) =
-                        warning_diags.filtered_source_diags_with_prefix(LINT_WARNING_PREFIX);
-                    report_warnings(&files, warning_diags);
-                    if any_linter_warnings {
-                        eprintln!("Please report feedback on the linter warnings at https://forums.sui.io\n");
-                    }
-                    if filtered_diags_num > 0 {
-                        eprintln!("Total number of linter warnings suppressed: {filtered_diags_num} (filtered categories: {filtered_categories})");
-                    }
+                    decorate_warnings(warning_diags, Some(&files));
                     fn_info = Some(Self::fn_info(&units));
                     Ok((files, units))
                 }
                 Err(error_diags) => {
+                    // with errors present don't even try decorating warnings output to avoid
+                    // clutter
                     assert!(!error_diags.is_empty());
-                    let any_linter_warnings = error_diags.any_with_prefix(LINT_WARNING_PREFIX);
-                    let (filtered_diags_num, filtered_categories) =
-                        error_diags.filtered_source_diags_with_prefix(LINT_WARNING_PREFIX);
                     let diags_buf = report_diagnostics_to_color_buffer(&files, error_diags);
-                    if let Err(err) = std::io::stderr().write_all(&diags_buf) {
+                    if let Err(err) = std::io::stdout().write_all(&diags_buf) {
```

Comment:
Why would we ever output diagnostics to stdout?

Reply:
I was reverting this piece to an earlier version and clearly got the code from the wrong place... In other words, yeah, there was no reason to change this to `stdout`...